packages/ecc-client-lit-ga4gh-tes/src/components/runs/runs.ts

```diff
@@ -0,0 +1,371 @@
+    {
+      tabGroup: "Output and Inputs",
+      children: [
+        {
+          label: "Output",
+          path: "outputs",
+        },
+        {
+          label: "Input",
+          path: "inputs",
+        },
+      ],
+    },
+  ];
+
+  @state() private filters: FilterProp[] = [
+    {
+      key: "search",
+      type: "search",
+      placeholder: "Search by prefix",
+    },
+    {
+      key: "tag",
+      type: "select",
+      options: [
+        "UNKNOWN",
+        "QUEUED",
+        "INITIALIZING",
+        "RUNNING",
+        "PAUSED",
+        "COMPLETE",
+        "EXECUTOR_ERROR",
+        "SYSTEM_ERROR",
+        "CANCELED",
+        "PREEMPTED",
+        "CANCELING",
+      ],
+      placeholder: "Filter by status",
+      selectConfig: {
+        multiple: true,
+      },
+    },
+  ];
+
+  @state() private items: ItemProp[] = [];
+  @state() private nextPageToken: string[] = [""];
+  @state() private view: "MINIMAL" | "BASIC" | "FULL" = "MINIMAL";
+  @state() private namePrefix = "";
+  @state() private cache = new Map();
+
+  private tagType: Record<string, string> = {
+    UNKNOWN: "neutral",
+    QUEUED: "warning",
+    INITIALIZING: "primary",
+    RUNNING: "primary",
+    PAUSED: "warning",
+    COMPLETE: "success",
+    EXECUTOR_ERROR: "danger",
+    SYSTEM_ERROR: "danger",
+    CANCELED: "danger",
+    PREEMPTED: "danger",
+    CANCELING: "danger",
+  };
+
+  protected updated(changedProperties: Map<PropertyKey, unknown>): void {
+    const eccUtilsDesignCollection = this.shadowRoot?.querySelector(
+      "ecc-utils-design-collection"
+      // Todo: Get the typeof Collections and use it instead of `any`
+    ) as any;
+    eccUtilsDesignCollection.pageSize = this.pageSize;
+    if (changedProperties.has("pageSize")) {
+      this._fetchData(1);
+    }
+
+    // Handle filter render
+    if (changedProperties.has("filter") || changedProperties.has("search")) {
+      this.filters = this.getUpdatedFilters();
+    }
+  }
+
+  private getUpdatedFilters(): FilterProp[] {
+    let updatedFilters = [...this.filters];
+
+    // Modify the array based on the conditions
+    if (!this.filter) {
+      updatedFilters = updatedFilters.filter(
+        (filter) => filter.type !== "select"
+      );
+    }
+    if (!this.search) {
+      updatedFilters = updatedFilters.filter(
+        (filter) => filter.type !== "search"
+      );
+    }
+
+    return updatedFilters;
+  }
+
+  private async _fetchData(page = 1) {
+    try {
+      // If all the items have been cached, don't invoke API call
+      if (this.nextPageToken === null) return;
+      const data = await fetchTasks(
+        this.baseURL,
+        this.pageSize,
+        this.nextPageToken[page - 1],
+        this.view,
+        this.namePrefix
+      );
+
+      const convertedData: ItemProp[] = [];
+      data.tasks?.forEach((task: { id: string; state: string }) => {
+        convertedData.push({
+          index: this.items.length + convertedData.length + 1,
```

Comment:
Put the index according to the items that have been fetched. See the logic for that here: https://github.com/elixir-cloud-aai/cloud-components/blob/main/packages/ecc-client-lit-ga4gh-wes/src/components/runs/runs.ts#L186

Also better to filter the items here!

Reply:
oh yeah thanks! :slightly_smiling_face: